step/step.go

```diff
@@ -3,15 +3,25 @@ package step
 import (
 	"fmt"
 	"path/filepath"
+	"regexp"
 	"strconv"
+	"strings"
 
 	"github.com/bitrise-io/go-steputils/v2/export"
 	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/pathutil"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
 	"github.com/bitrise-io/go-utils/v2/log"
 	"github.com/bitrise-io/go-xcode/v2/autocodesign/projectmanager"
 	"github.com/bitrise-io/go-xcode/xcodeproject/xcodeproj"
 )
 
+const (
+	infoPlistFileKey = "INFOPLIST_FILE"
+	envVarRegex      = `^.*\$\(.+\).*$`
```

Comment:
FYI I was curious and checked if it handles nested env vars. I'm not sure if it's even handled by Xcode, but it works 😄 
![image](https://github.com/bitrise-steplib/steps-set-xcode-build-number/assets/1694986/fc2872be-fb59-4430-adb9-b4450b6a1ff1)